src/Utility.cpp

```diff
@@ -11,17 +10,18 @@ std::u32string Gosu::utf8_to_composed_utc4(const std::string& utf8)
     utc4.reserve(utf8.size());
 
     const auto* current_byte = reinterpret_cast<const utf8proc_uint8_t*>(utf8.data());
-    auto remaining_length = utf8.length();
+    auto remaining_length = static_cast<utf8proc_ssize_t>(utf8.length());
 
     // First convert from UTF-8 to UTC-4.
     utf8proc_int32_t codepoint;
     while (remaining_length) {
         auto bytes_read = utf8proc_iterate(current_byte, remaining_length, &codepoint);
-        if (bytes_read == -1) {
+        if (bytes_read < 0) {
```

Comment:
FYI: This fixed bug #652. All while trying to improve test coverage... The magic of better tooling :)